### PR TITLE
Countdown Timer Edge App. 

### PR DESCRIPTION
**Pull request by salmanfarisvp**
Added Countdown Timer Edge App with features of 

- [X] Add timer over setting parameters.
- [X] Add timer heading over setting parameters. 
- [X] Timer works based on local time and we can override that with setting parameters. 
- [X] Local time displays for reference.

### CODE REVIEW

File: edge-apps/countdown-timer/index.html [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <title>Screenly Clock App</title>
```
Comment (vpetersson) [inline edge-apps/countdown-timer/index.html:5]: Change.

**Comment by vpetersson** [comment]
Please remove `edge-apps/countdown-timer/static/images/clock-app.jpg.

File: edge-apps/countdown-timer/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+### `timer`
+
+To set the countdown timer, we can use the `timer` settings.
+
+```bash
+$ Screenly edge-app setting set timer=2024-03-14T00:00:00
+# A relatively long console output...
+Edge app setting successfully set.
+```
+where `2024-03-14T00:00:00` your countdown timer limit with `YYYY-MM-DDTHH:MM:SS` format.
```
Comment (vpetersson) [inline edge-apps/countdown-timer/README.md:30]: It's important to point out that this is the countdown to a local timezone where the screen is. You can use `override_timezone` to pin a specific time zone across screens that span multiple time zones.

File: edge-apps/countdown-timer/screenly.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+user_version: v1.0.0
+description: Screenly Countdown Edge App
+icon: https://playground.srly.io/edge-apps/clock/static/images/icon.svg
```
Comment (vpetersson) [inline edge-apps/countdown-timer/screenly.yml:4]: Ask Adnan for an updated icon.

File: edge-apps/countdown-timer/screenly.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+author: Screenly, Inc.
+entrypoint: index.html
+settings:
+  enable_analytics:
+    type: string
+    default_value: 'true'
+    title: enable_analytics
```
Comment (vpetersson) [inline edge-apps/countdown-timer/screenly.yml:11]: Should be `Enable Analytics` (and fix this for all titles in the file).

File: edge-apps/countdown-timer/README.md (hunk 2 above)
Comment (salmanfarisvp) [reply edge-apps/countdown-timer/README.md:30]: Yeah, Currently it's reading the local timezone and the countdown is based on that.

File: edge-apps/countdown-timer/screenly.yml (hunk 4 above)
Comment (salmanfarisvp) [reply edge-apps/countdown-timer/screenly.yml]: Changed all the titles as per suggestion.

File: edge-apps/countdown-timer/screenly.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    type: string
+    default_value: GTM-P98SPZ9Z
+    title: Tag Manager Id
+    optional: true
+    help_text: Specify a Google Tag Manager ID
+  timer:
+    type: string
+    default_value: '2025-01-01T00:00:00'
+    title: Countdown Timer
+    optional: true
```
Comment (vpetersson) [inline edge-apps/countdown-timer/screenly.yml:36]: Definitely not optional.

Comment (salmanfarisvp) [reply edge-apps/countdown-timer/screenly.yml]: True. Changed.

File: edge-apps/countdown-timer/screenly.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    title: Override Timezone
+    optional: true
+    help_text: Specify a supported timezone value (e.g., Europe/London).
+  tag_manager_id:
+    type: string
+    default_value: GTM-P98SPZ9Z
+    title: Tag Manager Id
+    optional: true
+    help_text: Specify a Google Tag Manager ID
+  timer:
```
Comment (vpetersson) [inline edge-apps/countdown-timer/screenly.yml:32]: Let's rename this variable. It's not clear that this is the timestamp it is counting down to.

**Review by vpetersson — APPROVED** [review]
Approved pending:

* Manual testing two screens with two different time zones and a pinned `override_timezone` variable in the Edge App showing the same time
* Fixing the `timer` variable name.

File: edge-apps/countdown-timer/screenly.yml (hunk 6 above)
Comment (salmanfarisvp) [reply edge-apps/countdown-timer/screenly.yml]: Agree. Changed to `countdown_timer`. Is that ok?

Comment (vpetersson) [reply edge-apps/countdown-timer/screenly.yml]: ChatGPT suggests `countdown_to`, `countdown_end_time` or `taget_timestamp`. These are more descriptive of what the value actually is.

Comment (salmanfarisvp) [reply edge-apps/countdown-timer/screenly.yml]: Thanks for the suggestion, Selected `target_timestamp` as variable name.

**Review by vpetersson — APPROVED** [review]

File: edge-apps/countdown-timer/screenly.yml (hunk 3 above)
Comment (salmanfarisvp) [reply edge-apps/countdown-timer/screenly.yml]: Updated.  Thanks Adnan.

File: edge-apps/countdown-timer/README.md (hunk 2 above)
Comment (salmanfarisvp) [reply edge-apps/countdown-timer/README.md:30]: Tested and works fine.

**Comment by salmanfarisvp** [comment]
@vpetersson All checked passed, Can you merge this?  🎉 Thanks.

**Comment by salmanfarisvp** [comment]
@vpetersson Icon is added on https://github.com/Screenly/Playground/pull/157/commits/cefe0aa0ec88de47d68d8b0937f7d72ceaa5ed56 see  https://github.com/Screenly/Playground/blob/master/edge-apps/countdown-timer/static/images/icon.svg

**Comment by vpetersson** [comment]
We also need to deploy this from CI. Can you look at the other ones and leave the app_id as the placeholder for now so that I can fill that out when i get back to the desk.

**Comment by salmanfarisvp** [comment]
Sure. Do you want me to create a similar one right? https://github.com/Screenly/Playground/blob/master/.github/workflows/deploy-clock-app.yml .

**Comment by vpetersson** [comment]
Correct.

**Pull request merged**